pubsub/google/cloud/pubsub/message.py

```diff
@@ -85,7 +86,13 @@ def from_api_repr(cls, api_repr):
         :rtype: :class:`Message`
         :returns: The message created from the response.
         """
-        data = base64.b64decode(api_repr.get('data', b''))
+        try:
+            raw_data = api_repr.get('data', b'')
+            padded_data = raw_data + b'=' * (4 - len(raw_data) % 4)
```

Comment:
Have you tried running the pubsub system tests with this version of the patch?  Adding padding to a correctly-padded base64-encoded string might break things.  ISTM that we should try do decode as-is first, and only add padding if that fails.